tiledb/sm/group/group.cc

```diff
@@ -765,6 +765,18 @@ std::string Group::dump(
     const uint64_t num_indents,
     bool recursive,
     bool print_self) const {
+  // Create a set to track visited groups and prevent cycles
+  std::unordered_set<std::string> visited;
+  visited.insert(group_uri_.to_string());
+  return dump(indent_size, num_indents, recursive, print_self, visited);
+}
+
+std::string Group::dump(
+    const uint64_t indent_size,
+    const uint64_t num_indents,
+    bool recursive,
+    bool print_self,
+    std::unordered_set<std::string>& visited) const {
```

Comment:
It would be better if the helper function accepted `ss` as a parameter and write text to it, and returned void. This way, we would construct only one `std::string` at the end.